.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,35 @@
+name: Benchmark
+
+on:
+  push:
+    branchs: [ master ]
+  pull_request:
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: ['10.x', '12.x', '14.x']
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Cache NPM dependencies
+        uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: ${{ runner.OS }}-npm-cache
+          restore-keys: |
+            ${{ runner.OS }}-npm-cache
+      - name: Running benchmark
+        run: |
+          echo "============== Hexo Benchmark ==============" > benchmark-result-${{ matrix.node-version }}.txt
+          node test/benchmark.js >> benchmark-result-${{ matrix.node-version }}.txt
+      - uses: actions/upload-artifact@v2
+        with:
+          name: benchmark-results
+          path: benchmark-result-${{ matrix.node-version }}.txt
```

Comment:
We could post a comment at PR to show the benchmark result.

https://github.com/unsplash/comment-on-pr

@hexobot could be utilized.